src/vm/methodtable.cpp

```diff
@@ -3092,8 +3103,14 @@ void  MethodTable::AssignClassifiedEightByteTypes(SystemVStructRegisterPassingHe
                 currentEightByte++;
                 _ASSERTE(currentEightByte <= CLR_SYSTEMV_MAX_EIGHTBYTES_COUNT_TO_PASS_IN_REGISTERS);
 
-                currentEightByteOffset = offset + fieldSize;
-                accumulatedSizeForEightByte = 0;
+                currentEightByteOffset = offset + Min(fieldSize, (unsigned int)SYSTEMV_EIGHT_BYTE_SIZE_IN_BYTES);
+                accumulatedSizeForEightByte -= SYSTEMV_EIGHT_BYTE_SIZE_IN_BYTES;
+
+                // If a field is large enough to span multiple eightbytes, then set the eightbyte classification to the field's classification.
+                if (accumulatedSizeForEightByte > 0)
+                {
+                    helperPtr->eightByteClassifications[currentEightByte] = fieldClassificationType;
```

Comment:
We should set the eightBytesOffsets here too. In case our data structure filled the two eight bytes completely, we would not enter the `if` at line 3120 below and thus the `helperPtr->eightByteOffsets[currentEightByte]` would remain unset.